client/src/fuzzy-search.js

```diff
@@ -33,18 +25,11 @@ const getSubstrings = (regexString, matchString) => {
   let matchArray = regex.exec(matchString);
   let substrings = [];
   let isMatch = false;
-  let missingCount = 0;
-  let startsWith = 0;
   for (let i = 1; i < matchArray.length; i++) {
```

Comment:
since we don't need the index anymore, i'd go for `for (let match of matchArray) ...` instead